packages/server/modules/core/services/streams/streamAccessService.js

```diff
@@ -129,6 +131,13 @@ async function addOrUpdateStreamCollaborator(
 
   await validateStreamAccess(addedById, streamId, Roles.Stream.Owner)
 
+  // make sure server guests cannot be stream owners
+  if (role === Roles.Stream.Owner) {
+    const userServerRole = await ServerAcl.knex().where({ userId }).first()
```

Comment:
I'd prefer using the repository layer instead. You could re-use the `getStream()` function which also returns role metadata if you feed in a userId.